File: core/src/main/java/pcb/uwu/units/derived/fundamental/FrequencyUnit.java
package pcb.uwu.units.derived.fundamental;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.base.TimeUnit;

public class FrequencyUnit extends CompositeUnit {

	public FrequencyUnit(TimeUnit time) {
		super(new UnitCounter()
				.minor(time));
	}
}

File: dimension/src/main/java/pcb/uwu/unit/derived/acoustics/FrequencyUnit.kt
package pcb.uwu.unit.derived.acoustics

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.TimeUnit

open class FrequencyUnit : CompositeUnit
{
    @Suppress("ConvertSecondaryConstructorToPrimary")
    constructor(timeUnit: TimeUnit)
            : super(UnitCounter().minor(timeUnit))
}
